platform/openide.dialogs/src/org/openide/NotifyDescriptor.java

```diff
@@ -993,6 +994,25 @@ public Exception(Throwable detail, Object message) {
         }
     }
 
+    /**
+     * Notification providing a password input.
+     */
+    public static class PasswordLine extends InputLine {
```

Comment:
Please make the class `final` - we don't want random subclasses. And in case you ask: The fact that `InputLine` isn't final is more a design mistake than something done on purpose.